src/core/build_simulations.jl

```diff
@@ -77,21 +75,32 @@ function _build_stages(sim_ref::SimulationRef,
 
     system_to_file = get(kwargs, :system_to_file, true)
     mod_stages = Vector{_Stage}(undef, length(stages))
-
-    for (k, v) in stages
+    for (k, stage) in stages
```

Comment:
Since you aren't using k it would be better to write:
`for cache in values(stage.cache)`

Reply:
I am using the k to generate the paths.